recipes/openapi-generator/all/conanfile.py

```diff
@@ -0,0 +1,73 @@
+from conan import ConanFile
+from conan.tools.files import copy, download, save
+import os
+import stat
+
+
+required_conan_version = ">=1.47.0"
+
+
+class OpenApiGeneratorConan(ConanFile):
+    name = "openapi-generator"
+    description = "Generation of API client libraries (SDK generation), server stubs, documentation and configuration automatically given an OpenAPI Spec (v2, v3)"
+    license = "Apache-2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://openapi-generator.tech"
+    topics = ("api", "sdk", "generator", "openapi")
+    settings = "os", "arch", "compiler", "build_type"
+
+    def layout(self):
+        pass
+
+    def package_id(self):
+        del self.info.settings.arch
+        del self.info.settings.compiler
+        del self.info.settings.build_type
+
+    def source(self):
+        pass
+
+    def build(self):
+        v = self.conan_data["sources"][self.version]
+        download(
+                self,
+                url=v["url"],
+                filename=os.path.join(self.source_folder, "openapi-generator.jar"),
+                sha256=v["sha256"],
+                )
+        download(
+                self,
+                url="https://raw.githubusercontent.com/OpenAPITools/openapi-generator/master/LICENSE",
+                filename=os.path.join(self.source_folder, "LICENSE"),
+                sha256="91a2fcdfc23cbd1188a22cc3b76647bf6eb05c87889e376a19fe478f0398ff02",
+                )
+
+    def package(self):
+        copy(self, pattern="LICENSE", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
+        copy(self, pattern="openapi-generator.jar", dst=os.path.join(self.package_folder, "res"), src=self.source_folder)
+        jar = os.path.join(self.package_folder, "res", "openapi-generator.jar")
+        if self.info.settings.os == "Windows":
+            save(self,
+                 path=os.path.join(self.package_folder, "bin", "openapi-generator.bat"),
+                 content=f"""\
+                         java -jar {jar} %*
+                         """
+                 )
+        else:
+            bin_path = os.path.join(self.package_folder, "bin", "openapi-generator")
+            save(self,
+                 path=bin_path,
+                 content=f"""\
+                         #!/bin/bash
+                         java -jar {jar} $@
```

Comment:
What is java is not installed 🤔

Reply:
This thought did occur to me, too. I wasn't sure if perhaps we had any other Java packages around. Would this require a `java/system` package?

Reply:
Are not openjdk or openjdk-zulu packages enough?

Reply:
Had no idea we had an `openjdk` package. Nice!

Reply:
I would say this should have a require openjdk for it to make the most sense 😉 that's a great solution

(insert some meme about C++ being better than Java -- enjoy my bad jokes)

Reply:
I thought I would need to add openjdk as a requirement but I was confused how to do it: adding a tool_requires in requirements method ?)

Since the CI pass I decide to wait for your comment.

![](https://i.makeagif.com/media/10-26-2015/UTNcmT.gif)

Reply:
Do you need java for runtime? If yes, add as regular self.requires in requirements(). Otherwise, tool_requires in build_requirements().